plugins/plant.cpp

```diff
@@ -0,0 +1,627 @@
+// Grow and remove shrubs or trees.
+
+#include <iostream>
+#include <iomanip>
+#include <map>
+#include <algorithm>
+#include <vector>
+#include <string>
+
+#include "Debug.h"
+#include "Core.h"
+#include "Console.h"
+#include "Export.h"
+#include "LuaTools.h"
+#include "PluginManager.h"
+#include "TileTypes.h"
+
+#include "modules/Gui.h"
+#include "modules/MapCache.h"
+#include "modules/Maps.h"
+
+#include "df/block_square_event_grassst.h"
+#include "df/map_block.h"
+#include "df/map_block_column.h"
+#include "df/plant.h"
+#include "df/plant_raw.h"
+#include "df/world.h"
+
+using std::vector;
+using std::string;
+using namespace DFHack;
+
+DFHACK_PLUGIN("plant");
+REQUIRE_GLOBAL(world);
+
+namespace DFHack
+{
+    DBG_DECLARE(plant, log, DebugCategory::LINFO);
+}
+
+struct cuboid
```

Comment:
I'll keep this in mind for refactoring if we want to make it more widely accessible. seems like it should be useful to many map-related tools